rewrite-gradle/src/test/java/org/openrewrite/gradle/AddGradleWrapperTest.java

```diff
@@ -108,6 +101,67 @@ void addWrapper() {
         );
     }
 
+
+    @Test
+    void addWrapperWithReleaseTag() {
+
+        AtomicReference<String> output = new AtomicReference<>("");
+        rewriteRun(
+          spec ->
+            spec
+              .recipe(
+                //language=yaml
+                new ByteArrayInputStream(
```

Comment:
With my other comment, this simply becomes creating the recipe with all null constructor arguments.

Reply:
Wouldn't this be confusing? 

And, would we expect people to pass them while building them? 

With the current setup, it seems that the recipe cannot be executed from the command line as it needs this formal configuration.

Reply:
So the short answer, is yes. The build plugins both understand required vs optional parameters, so if all parameters are optional then they are able to just provide null or empty values. This enables construction of recipes that have many parameters that are all optional or of recipes which have some required and some optional without having to specify each of those parameters.

Reply:
I think it might be easier to understand this if we were to use the java constructor to create these instances. This also helps to act as a verifiable test for my other comment as it seems the YAML recipe version is coercing the blanks to null which doesn't happen everywhere the same way.